libs/framework/include/celix/dm/ServiceDependency.h

```diff
@@ -59,20 +61,20 @@ namespace celix { namespace dm {
             }
         }
     public:
-        BaseServiceDependency(bool v)  : valid{v} {
+        BaseServiceDependency(celix_dm_component_t* c, bool v)  : cCmp{c}, valid{v} {
             if (this->valid) {
                 this->cServiceDep = celix_dmServiceDependency_create();
                 //NOTE using suspend as default strategy
                 celix_dmServiceDependency_setStrategy(this->cServiceDep,  DM_SERVICE_DEPENDENCY_STRATEGY_SUSPEND);
             }
         }
 
-        virtual ~BaseServiceDependency() = default;
+        virtual ~BaseServiceDependency() noexcept;
```

Comment:
Fun fact: implicit noexcept on destructors behave differently than on functions. For normal functions, not specifying noexcept results in the compiler adding `noexcept(false)`, which means that the compiler should assume exceptions can potentially be thrown and some optimizations should not be done (though I would not be surprised if compilers check if a function throws and optimizes anyway.)

Noexcept on destructors are defined differently: if it is missing, the compiler should check if all its parent class and all member variables have noexcept destructors. If so, mark destructor implicitly as `noexcept(true)` otherwise `noexcept(false)`.

Something similar happens for default defined move/copy/regular constructors/operators. If you happen to declare a copy constructor without explicitly telling the compiler if it is noexcept or not, it will try to figure it out for you.

Given that `BaseServiceDependency` has no throwing parents or member variables, it was already marked noexcept. Though I like explicit noexcept. So just wanted to share this tidbit of knowledge.

Reply:
:+1: